topology/intel/dmic-nhlt.c

```diff
@@ -0,0 +1,1632 @@
+/*
+  Copyright(c) 2021 Intel Corporation
+  All rights reserved.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of version 2 of the GNU General Public License as
+  published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St - Fifth Floor, Boston, MA 02110-1301 USA.
+  The full GNU General Public License is included in this distribution
+  in the file called LICENSE.GPL.
+*/
+
+#include <stdint.h>
+#include <errno.h>
+#include <stdio.h>
+#include <alsa/input.h>
+#include <alsa/output.h>
+#include <alsa/conf.h>
+#include <alsa/error.h>
+#include "pre-processor.h"
+#include "dmic-nhlt.h"
+#include "dmic/pdm_decim_table.h"
+#include "dmic/dmic_macros.h"
+
+/* Structs for gathering the parameters from topology */
+struct dmic_config_pdm {
+        uint16_t id;
+        uint16_t enable_mic_a;
+        uint16_t enable_mic_b;
+        uint16_t polarity_mic_a;
+        uint16_t polarity_mic_b;
+        uint16_t clk_edge;
+        uint16_t skew;
+};
+
+struct dmic_config_dai {
+	uint32_t driver_version;
+	uint32_t io_clk;
+        uint32_t pdmclk_min;
+        uint32_t pdmclk_max;
+        uint32_t fifo_fs;
+        uint16_t fifo_bits;
+        uint16_t fifo_bits_b;
+        uint16_t duty_min;
+        uint16_t duty_max;
+        uint32_t num_pdm_active;
+        uint32_t wake_up_time;
+        uint32_t min_clock_on_time;
+        uint32_t unmute_ramp_time;
+        struct dmic_config_pdm pdm[SOF_DAI_INTEL_DMIC_NUM_CTRL];
```

Comment:
WHY SOF? This is not dependent on any SOF definitions or choices, and not even on SOF use.

Reply:
removed